runtime/src/snapshot_package/compare.rs

```diff
@@ -40,21 +40,53 @@ pub fn cmp_snapshot_archive_kinds_by_priority(
     }
 }
 
+/// Check if two snapshots are the same kind
+#[must_use]
+pub fn are_snapshots_the_same_kind(a: &SnapshotPackage, b: &SnapshotPackage) -> bool {
+    are_snapshots_kinds_the_same_kind(&a.snapshot_kind, &b.snapshot_kind)
+}
+
+/// Check if two snapshot kinds are the same kind.
+#[must_use]
+pub fn are_snapshots_kinds_the_same_kind(a: &SnapshotKind, b: &SnapshotKind) -> bool {
+    use SnapshotKind as Kind;
+    match (a, b) {
+        (Kind::Archive(archive_a), Kind::Archive(archive_b)) => {
+            are_snapshot_archives_kinds_the_same_kind(archive_a, archive_b)
+        }
+    }
+}
+
+// Check if two snapshot archives are the same kind. Incremental snapshot archives with different
+// base slots are considered the same kind.
```

Comment:
nit: /// doc comments, and then moved the additional details part down to newline.

```suggestion
/// Check if two snapshot archives are the same kind
///
/// Incremental snapshot archives with different base slots are considered the same kind.
```